github/githubapp.go

```diff
@@ -17,18 +19,38 @@ type GitHubApp struct {
 	ClientId       string `json:"client_id"`
 	InstallationId int    `json:"installation_id"`
 	PrivateKey     string `json:"private_key"` // SENSITIVE
+	parsedRSAKey   *rsa.PrivateKey
```

Comment:
I really don't like the fact that this is nill unless someone calls `app.Validate()` method first.

Reply:
This is normal life with pointers.
The real question is: why do we need to use a pointer here?

Reply:
Ping

Reply:
These are not super valid reasons but:
1. it seemed as a good "fit" because I could just do: 
```
key, err := jwt.ParseRSAPrivateKeyFromPEM([]byte(app.PrivateKey))
...
app.parsedRSAKey = key
```
since `jwt.ParseRSAPrivateKeyFromPEM` returns a pointer.
3. it allow me to easily check for empty struct with: `*app == GitHubApp{}`
Now, I had to write a more "complex" function to do that.

Anyway... none of this is a good enough explanation and I am happy to hear your opinion on when to use one or the other approach...